### PR TITLE
[CMA] Index components via Orient DB hooks

### PR DESCRIPTION
**Pull request by adreghiciu**
Moving to a hook it will ensure that indexing of components will always be run and eliminate the need for repository implementation to call into indexing

### CODE REVIEW

**Comment by kellyrob99** [comment]
+1

**Comment by cwilper** [comment]
+1

**Comment by mrprescott** [comment]
+1 I like it, I like that it's low level so it can't miss anything, but it's controlled by storage. It's okay that we're indexing assets as well as components?

**Comment by cstamas** [comment]
+1

**Comment by adreghiciu** [comment]
@mrprescott we are only index on component type events, so we do nothing when assets change, which may be problematic if we update some asset without changing the component in any way

**Comment by mrprescott** [comment]
@adreghiciu Got it. I missed the filtering on V_COMPONENT. We may need to revisit for maven.

**Comment by jdillon** [comment]
Curious, why?  Was previous means/api to manage the index not working?

File: components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java [hunk 1]
```diff
@@ -85,6 +96,16 @@ protected void doInit() throws Exception {
     super.doInit();
   }
 
+  @Override
+  protected void doStart() throws Exception {
+    try {
+      searchFacet = getRepository().facet(SearchFacet.class);
+    }
+    catch (MissingFacetException e) {
+      // no search facet, no indexing
```
Comment (jdillon) [inline components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageFacetImpl.java]: are there real cases where a repo would not have a search facet?  ie is the nullability here causing additional logic/code to manage a situation that likely will never happen?

**Comment by adreghiciu** [comment]
Indexing was all working fine, but you have to call into indexing facet in different places.
Moving this to a hook will ensure that indexing is always done regardless from where the components are changed.
As we also figured out you have to call after transaction is committed as the RID we used is not set if transaction is not committed.

**Comment by adreghiciu** [comment]
Manually merged into nexus-cma3 https://github.com/sonatype/nexus-oss/commit/4e589cb344cadf0267bbd42b7b721807ddc7b3ac

**Pull request closed without merging**